src/middlewared/middlewared/main.py

```diff
@@ -1106,14 +1116,37 @@ def register_hook(self, name, method, *, inline=False, order=0, raise_error=Fals
 
         self.__hooks[name].append({
             'method': method,
+            'blockable': blockable,
             'inline': inline,
             'order': order,
             'raise_error': raise_error,
             'sync': sync,
         })
         self.__hooks[name] = sorted(self.__hooks[name], key=lambda hook: hook['order'])
 
+    @contextlib.contextmanager
```

Comment:
On second thought, does it make sense to do it like this so we eliminate a `for` loop iteration?
```
    @contextlib.contextmanager
    def block_hooks(self, *names):
        with self.__blocked_hooks_lock:
            for name in names:
                if not self.__hooks[name]:
                    raise RuntimeError(f'Hook {name!r} does not exist')
                elif not self.__hooks[name][0]['blockable']:
                    raise RuntimeError(f'Hook {name!r} is not blockable')
                else:
                    self.__blocked_hooks[name] += 1
        yield

        with self.__blocked_hooks_lock:
            for name in names:
                self.__blocked_hooks[name] -= 1
```